tiledb/sm/serialization/consolidation.cc

```diff
@@ -53,36 +54,76 @@ namespace tiledb {
 namespace sm {
 namespace serialization {
 
+class ConsolidationSerializationException : public StatusException {
+ public:
+  explicit ConsolidationSerializationException(const std::string& message)
+      : StatusException("[TileDB::Serialization][Consolidation]", message) {
+  }
+};
+
 #ifdef TILEDB_SERIALIZATION
 
-Status array_consolidation_request_to_capnp(
+void array_consolidation_request_to_capnp(
     const Config& config,
+    const std::vector<std::string>* fragment_uris,
     capnp::ArrayConsolidationRequest::Builder*
         array_consolidation_request_builder) {
+  // Validate input arguments to be sure that what we are serializing make sense
+  auto mode = Consolidator::mode_from_config(config);
+  if (mode != ConsolidationMode::FRAGMENT &&
+      (fragment_uris != nullptr || !fragment_uris->empty())) {
```

Comment:
The LHS of this OR evaluates to false if `fragment_uris` is null so the RHS is still checked and that segfaults in existing consolidation tests not changed in this PR

```suggestion
      (fragment_uris != nullptr && !fragment_uris->empty())) {
```

Reply:
ah a classic one, thanks!